controllers/machinehealthcheck_controller.go

```diff
@@ -299,10 +299,11 @@ func (r *MachineHealthCheckReconciler) PatchHealthyTargets(ctx context.Context,
 			// Get remediation request object
 			obj, err := r.getExternalRemediationRequest(ctx, m, t.Machine.Name)
 			if err != nil {
-				if apierrors.IsNotFound(errors.Cause(err)) {
-					continue
+				if !apierrors.IsNotFound(errors.Cause(err)) {
+					logger.Error(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
+					errList = append(errList, errors.Wrapf(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName))
```

Comment:
you could do something like this, to avoid duplicating the message:

```suggestion
					wrappedErr := errors.Wrapf(err, "failed to fetch remediation request for machine %q in namespace %q within cluster %q", t.Machine.Name, t.Machine.Namespace, t.Machine.ClusterName)
					logger.Error(wrappedErr)
					errList = append(errList, wrappedErr)
```

Reply:
updated , thks